### ISSUE TITLE
s///g incorrectly returns 0 with $^R on right side

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4490]
Migrated from [rt.perl.org#7804](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=7804) (status was 'resolved')

Searchable as RT7804$

**Comment by p5pRT** [comment]
### From @rgarcia



This program :

$\_ = 'abc';
$s = s\{ a \(?\{ 'A' \}\) | b \(?\{ 'B' \}\) \}\{$^R\}xg;
print "\<$\_> \<$s>\\n";

outputs : \(with perl 5\.7\.2\)
\<ABc> \<0>
But this should output \<ABc> \<2> instead\.

If you remove the 'g'\, the correct result is given : \<Abc> \<1>
If you replace the $^R by another string\, the correct result is given also\.
Stranger : if you replace $^R by $^X\, or by another $^\.\.\. variable\, the
result is incorrect and $s becomes == 0\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.7.2:

Configured by rafael at Tue Oct  9 10:10:11 CEST 2001.

Summary of my perl5 (revision 5.0 version 7 subversion 2 patch 12374) configuration:
  Platform:
    osname=linux, osvers=2.2.12-20, archname=i686-linux-thread-multi
    uname='linux rafael.kazibao.net 2.2.12-20 #1 mon sep 27 10:40:35 edt 1999 i686 unknown '
    config_args='-des -Dusedevel -Dprefix=/opt/perl -Dusethreads'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2',
    cppflags='-D_REENTRANT -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='egcs-2.91.66 19990314/Linux (egcs-1.1.2 release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldb -ldl -lm -lpthread -lc -lposix -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lpthread -lc -lposix -lcrypt -lutil
    libc=/lib/libc-2.1.3.so, so=so, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    DEVEL12362


@INC for perl v5.7.2:
    /home/rafael/perllib
    /opt/perl/lib/5.7.2/i686-linux-thread-multi
    /opt/perl/lib/5.7.2
    /opt/perl/lib/site_perl/5.7.2/i686-linux-thread-multi
    /opt/perl/lib/site_perl/5.7.2
    /opt/perl/lib/site_perl
    .


Environment for perl v5.7.2:
    HOME=/home/rafael
    LANG=en_US
    LANGUAGE (unset)
    LC_ALL=en_US
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/rafael/bin:/usr/local/bin:/bin:/usr/bin:/usr/X11R6/bin:/usr/local/apache/bin:/usr/local/java/jdk1.3.0_02/bin
    PERLLIB=/home/rafael/perllib
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @cwest

resolved by \#20684

**Comment by p5pRT** [comment]
@cwest - Status changed from 'open' to 'resolved'
